workflow/util/util.go

```diff
@@ -827,6 +827,29 @@ func resetConnectedParentGroupNodes(oldWF *wfv1.Workflow, newWF *wfv1.Workflow,
 	return newWF, resetParentGroupNodes
 }
 
+// RteryWorkflow retry a workflow by setting spec.retry and controller will retry it.
+func RetryWorkflow(ctx context.Context, wf *wfv1.Workflow, restartSuccessful bool, nodeFieldSelector string, parameters []string) (*wfv1.Workflow, error) {
```

Comment:
We could use a better name here, `retryworkflow` can be confusing,
The word on top of my head is `markWorkflowForRetry`, there must be a better term

Reply:
Changed, Thanks!